packages/vitest-plugin-storybook/src/global-setup.ts

```diff
@@ -0,0 +1,81 @@
+import { spawn, type ChildProcess } from 'node:child_process'
+import { log } from './utils'
+import http from 'node:http'
+import https from 'node:https'
+
+import type { GlobalSetupContext } from 'vitest/node'
+
+let storybookProcess: ChildProcess | null = null
+
+const checkStorybookRunning = (storybookUrl: string) => {
+  return new Promise<boolean>((resolve) => {
+    const url = new URL(`${storybookUrl}/iframe.html`)
+    const protocol = url.protocol === 'https:' ? https : http
+
+    const options = {
+      method: 'HEAD',
+      host: url.hostname,
+      path: url.pathname,
+    }
+
+    const req = protocol.request(options, (res) => {
+      resolve(res.statusCode === 200)
+    })
+
+    req.on('error', () => resolve(false))
+    req.end()
```

Comment:
why not just fetch?

```ts
const checkStorybookRunning = async (storybookUrl: string): Promise<boolean> => {
  try {
    const response = await fetch(`${storybookUrl}/iframe.html`, { method: 'HEAD' });
    return response.ok;
  } catch {
    return false;
  }
};
```

Reply:
Fetch isn't natively available in Node <21, so I get that. It's overkill to pull in node-fetch or something like that for a simple HEAD request.

Reply:
I thought it was available since Node 18?

<img width="814" alt="image" src="https://github.com/user-attachments/assets/7542def2-2331-48ad-bbbf-1fd629e21301">

Reply:
I thought of doing this fetch initially, but ended up using http for compatibility purposes. I am ok with using fetch if that works from Node 18 onwards. I will experiment!

Reply:
Seems like in node 18 you **can** use fetch natively without the need of running it via `--experimental-fetch`, weird! I thought you needed it until Node 21. I'll make this change and we can revisit if needed.